modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -3066,6 +3082,12 @@ private Expression<?>[] _getSelectExpressions(
 		for (ObjectField objectField :
 				dynamicObjectDefinitionTable.getObjectFields()) {
 
+			if ((selectedFields != null) &&
```

Comment:
It will be faster if we do this check below this portion:

			if (!objectField.compareBusinessType(
					ObjectFieldConstants.BUSINESS_TYPE_AGGREGATION) &&
				!objectField.compareBusinessType(
					ObjectFieldConstants.BUSINESS_TYPE_FORMULA)) {
				continue;
			}